src/ray/raylet/worker_pool_test.cc

```diff
@@ -346,5 +346,6 @@ TEST_F(WorkerPoolTest, StartWorkerWithDynamicOptionsCommand) {
 
 int main(int argc, char **argv) {
   ::testing::InitGoogleTest(&argc, argv);
+  RayConfig::instance().initialize({{"enable_multi_tenancy", "true"}});
```

Comment:
Can we add some unit tests for multitenancy? Maybe:
1. Make sure PopWorker returns the right type of worker when there are multiple jobs.
2. Making sure the callback for creating a driver only gets called once all the initial workers have started (assuming we move that code into the WorkerPool and multitenancy is disabled).

Also, can we also add cases/run the suite for enable_multi_tenancy=false? Especially considering this is the default.

Reply:
1. Yeah, I can add that.
2. I assume you were referring to the `send_reply_callback` of the `RegisterDriver` method. If multi-tenancy is disabled, the callback will be invoked immediately and the old logic of spawning initial workers in the constructor `WorkerPool` for every job is kept.

I'll try to make it testable for both true and false of the `enable_multi_tenancy` flag.

Reply:
@stephanie-wang If multi-tenancy is disabled, the driver's `ray.init()` will not wait for the initial workers to be started. But for other reasons, it looks like it waited. See [here](https://github.com/ray-project/ray/pull/9500#issuecomment-659791940) for the explanation. Do you want me to change the behavior to always wait for initial workers no matter whether multi-tenancy is enabled or not?

Reply:
I've added a test case for 1) and make worker pool test available for both enable_multi_tenancy=true and enable_multi_tenancy=false.

Reply:
That sounds good to me! It's fine if we don't cover case 2.

Reply:
BTW, case 2 with multi-tenancy on is covered by `test_initial_workers` in `test_multi_tenancy.py`.